### PR TITLE
major update of plugin to QGIS3

### PR DESCRIPTION
**Pull request by landam**
Based on https://github.com/davidtethal/gisquick-qgis-plugin-gsoc2018). Some issues needs to be solved before merging into master.

### CODE REVIEW

File: webgisplugin.py [hunk 1]
```diff
@@ -285,80 +284,76 @@ def layers_list(self):
...
-        # build complete layers tree
-        layers_root = self._get_project_layers_tree()
 
-        # filter to base layers only
-        base_layers = {
-            layer.id(): layer
-            for layer in self.layers_list()
-                if self.is_base_layer_for_publish(layer)
-        }
+        def overlays_tree(tree_node):
```
Comment (landam) [inline webgisplugin.py]: @davidtethal Please review this part.

File: webgisplugin.py [hunk 2]
```diff
@@ -285,80 +284,76 @@ def layers_list(self):
...
-                overlay_child = overlays_tree(child)
-                if overlay_child:
-                    overlay_children.append(overlay_child)
-            if overlay_children:
-                return Node(node.name, overlay_children)
-            elif not node.children and node.name in overlay_layers:
-                return Node(node.name, layer=overlay_layers[node.name])
-
-        return overlays_tree(layers_root)
+        def overlays_tree(tree_node):
```
Comment (landam) [inline webgisplugin.py:354]: @davidtethal Please review this part.

File: webgisplugin.py (hunk 1 above)
Comment (marcel-dancak) [reply webgisplugin.py]: It's quite possible that I wrote this piece of code, and amount of changes are due to changes in Qgis api. There was iface.legendInterface().groupLayerRelationship() in Qgis 2, which was removed and instead iface.layerTreeView().layerTreeModel() was added.

File: webgisplugin.py (hunk 2 above)
Comment (landam) [reply webgisplugin.py:354]: The problem was code duplication. Two almost same `overlays_tree()` methods. Solved by 81e08be

File: webgisplugin.py (hunk 1 above)
Comment (landam) [reply webgisplugin.py]: The problem was code duplication. Two almost same overlays_tree() methods. Solved by 81e08be

**Comment by landam** [comment]
I no objection for reviewers (@jachym @davidtethal @marcel-dancak ) I will merge this branch into master and prepare for new plugin release 2.0.0.

**Comment by marcel-dancak** [comment]
For this goal, which is mostly about update of Python version and QGIS API, it looks fine. I can see several drawbacks of original design, but that's not relevant here.

**Pull request merged**